cmd/compgen/cmd/common/common.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type TemplateData struct {
+	Name        string
+	PackageName string
+}
+
+// Executes `go mod tidy`
+func Tidy(path string) {
+	cmd := exec.Command("go", "mod", "tidy", "-e")
+	cmd.Dir = path
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		panic(string(output))
```

Comment:
It's generally not good practice to panic. I would recommend using a logger to log the error and exit, but not panic.

Reply:
This used to be log.Fatal which calls os.Exit(1) after logging a message. That wasn't testable, hence the change to panic(). The idea with these statements is that compgen should stop executing.

Reply:
Fatal & errors should be testable. Panics should be used when something impossible happens, literally the application dying. Panics show the stack and are much harder to troubleshoot. Let's avoid them please. https://go.dev/doc/effective_go#panic